File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans=[]
        def backtrack(arr,pointer):
            if sum(arr)>target:
                return
            elif sum(arr)==target:
                ans.append(arr)            
            while pointer<len(nums):
                arr.append(nums[pointer])
                backtrack(arr.copy(),pointer)
                arr.pop()
                pointer+=1
        backtrack([],0)
        return ans
        

File: Data Structures & Algorithms/majority-element-ii/submission-3.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int target=nums.length/3;
        List<Integer> ans=new ArrayList<>();
        Map<Integer, Integer> map=new HashMap<>();
        for (int num:nums){
            map.merge(num,1,Integer::sum);
        }
        for(Map.Entry<Integer, Integer> e: map.entrySet()){
            if (e.getValue()>target){
                ans.add(e.getKey());
            }
        }
        return ans;
    }
}